--- geos-2.0.0/tools/geos-config.in	2004-07-17 09:07:17.000000000 +0000
+++ geos-2.0.0.new/tools/geos-config.in	2004-09-24 21:25:31.312553288 +0000
@@ -31,10 +31,10 @@
     echo @VERSION@
      ;;
     --cflags)
-    echo @CFLAGS@ -I@prefix@/include 
+    echo -I@prefix@/include
       ;;
     --libs)
-      echo -L@prefix@/lib -lgeos
+      echo -L@libdir@ -lgeos
       ;;
     --includes)
       echo @prefix@/include
