--- geos-3.2.2/macros/ac_pkg_swig.m4.orig	2007-08-30 08:19:41.000000000 +0200
+++ geos-3.2.2/macros/ac_pkg_swig.m4	2010-10-09 13:30:47.923740205 +0200
@@ -67,9 +67,9 @@
                         if test -z "$available_patch" ; then
                                 [available_patch=0]
                         fi
-                        if test $available_major -ne $required_major \
-                                -o $available_minor -ne $required_minor \
-                                -o $available_patch -lt $required_patch ; then
+                        if test $available_major -lt $required_major \
+                                -o $available_major -eq $required_major -a $available_minor -lt $required_minor \
+                                -o $available_major -eq $required_major -a $available_minor -eq $required_minor -a $available_patch -lt $required_patch ; then
                                 AC_MSG_WARN([SWIG version >= $1 is required.  You have $swig_version.  You should look at http://www.swig.org])
                                 SWIG=''
                         else
