--- geos-3.3.7/configure.in.orig	2013-01-22 19:21:11.000000000 +0100
+++ geos-3.3.7/configure.in	2013-01-26 13:00:57.162086197 +0100
@@ -37,8 +37,7 @@
 
 AM_INIT_AUTOMAKE([geos], [$VERSION], [no-define])
 AM_MAINTAINER_MODE
-AM_CONFIG_HEADER([include/config.h])
-AM_CONFIG_HEADER([include/geos/platform.h])
+AC_CONFIG_HEADERS([include/config.h include/geos/platform.h])
 AC_PROG_CC
 
 dnl use libtool ----------------------------------------------------------
