diff -urN geos-3.0.0/doc/example.cpp geos-3.0.0.new/doc/example.cpp
--- geos-3.0.0/doc/example.cpp	2007-12-12 22:59:31.000000000 +0100
+++ geos-3.0.0.new/doc/example.cpp	2008-10-18 15:27:34.000000000 +0200
@@ -28,6 +28,7 @@
  *
  **********************************************************************/
 
+#include <cstdlib>
 #include <vector>
 #include <sstream>
 #include <iomanip>
diff -urN geos-3.0.0/source/geomgraph/EdgeNodingValidator.cpp geos-3.0.0.new/source/geomgraph/EdgeNodingValidator.cpp
--- geos-3.0.0/source/geomgraph/EdgeNodingValidator.cpp	2007-12-12 22:59:26.000000000 +0100
+++ geos-3.0.0.new/source/geomgraph/EdgeNodingValidator.cpp	2008-10-18 15:21:38.000000000 +0200
@@ -18,10 +18,10 @@
 
 #include <vector>
 
+#include <geos/geom/CoordinateSequence.h>
 #include <geos/geomgraph/EdgeNodingValidator.h>
 #include <geos/geomgraph/Edge.h>
 #include <geos/noding/SegmentString.h>
-#include <geos/geom/CoordinateSequence.h>
 
 using namespace std;
 using namespace geos::noding;
diff -urN geos-3.0.0/source/geomgraph/GeometryGraph.cpp geos-3.0.0.new/source/geomgraph/GeometryGraph.cpp
--- geos-3.0.0/source/geomgraph/GeometryGraph.cpp	2007-12-12 22:59:26.000000000 +0100
+++ geos-3.0.0.new/source/geomgraph/GeometryGraph.cpp	2008-10-18 15:22:16.000000000 +0200
@@ -42,6 +42,7 @@
 
 #include <geos/inline.h>
 
+#include <typeinfo>
 #include <vector>
 #include <memory> // auto_ptr
 #include <cassert>
diff -urN geos-3.0.0/source/headers/geos/noding/IntersectionAdder.h geos-3.0.0.new/source/headers/geos/noding/IntersectionAdder.h
--- geos-3.0.0/source/headers/geos/noding/IntersectionAdder.h	2007-12-12 22:59:29.000000000 +0100
+++ geos-3.0.0.new/source/headers/geos/noding/IntersectionAdder.h	2008-10-18 15:06:02.000000000 +0200
@@ -19,6 +19,7 @@
 #include <vector>
 #include <iostream>
 #include <cmath> // for abs()
+#include <cstdlib>
 
 #include <geos/inline.h>
 
diff -urN geos-3.0.0/source/index/quadtree/DoubleBits.cpp geos-3.0.0.new/source/index/quadtree/DoubleBits.cpp
--- geos-3.0.0/source/index/quadtree/DoubleBits.cpp	2007-12-12 22:59:26.000000000 +0100
+++ geos-3.0.0.new/source/index/quadtree/DoubleBits.cpp	2008-10-18 14:59:01.000000000 +0200
@@ -21,7 +21,7 @@
 #include <geos/index/quadtree/DoubleBits.h>
 #include <geos/util/IllegalArgumentException.h>
 
-#include <string>
+#include <cstring>
 
 #if __STDC_IEC_559__
 #define ASSUME_IEEE_DOUBLE 1
diff -urN geos-3.0.0/source/io/ByteOrderValues.cpp geos-3.0.0.new/source/io/ByteOrderValues.cpp
--- geos-3.0.0/source/io/ByteOrderValues.cpp	2007-12-12 22:59:26.000000000 +0100
+++ geos-3.0.0.new/source/io/ByteOrderValues.cpp	2008-10-18 15:01:27.000000000 +0200
@@ -22,6 +22,7 @@
 #include <geos/util.h>
 
 #include <cassert>
+#include <cstring>
 
 namespace geos {
 namespace io { // geos.io
diff -urN geos-3.0.0/source/io/StringTokenizer.cpp geos-3.0.0.new/source/io/StringTokenizer.cpp
--- geos-3.0.0/source/io/StringTokenizer.cpp	2007-12-12 22:59:26.000000000 +0100
+++ geos-3.0.0.new/source/io/StringTokenizer.cpp	2008-10-18 15:01:01.000000000 +0200
@@ -16,6 +16,7 @@
 
 #include <geos/io/StringTokenizer.h>
 
+#include <cstdlib>
 #include <string>
 
 using namespace std;
diff -urN geos-3.0.0/source/noding/FastNodingValidator.cpp geos-3.0.0.new/source/noding/FastNodingValidator.cpp
--- geos-3.0.0/source/noding/FastNodingValidator.cpp	2007-12-12 22:59:30.000000000 +0100
+++ geos-3.0.0.new/source/noding/FastNodingValidator.cpp	2008-10-18 15:04:49.000000000 +0200
@@ -17,12 +17,12 @@
  *
  **********************************************************************/
 
-#include <geos/noding/FastNodingValidator.h>
 #include <geos/noding/MCIndexNoder.h> // for checkInteriorIntersections()
 #include <geos/noding/SingleInteriorIntersectionFinder.h>
 #include <geos/util/TopologyException.h> // for checkValid()
 #include <geos/geom/Coordinate.h>
 #include <geos/io/WKTWriter.h> // for getErrorMessage()
+#include <geos/noding/FastNodingValidator.h>
 
 #include <string>
 #include <iostream>
diff -urN geos-3.0.0/source/operation/buffer/OffsetCurveSetBuilder.cpp geos-3.0.0.new/source/operation/buffer/OffsetCurveSetBuilder.cpp
--- geos-3.0.0/source/operation/buffer/OffsetCurveSetBuilder.cpp	2007-12-12 22:59:25.000000000 +0100
+++ geos-3.0.0.new/source/operation/buffer/OffsetCurveSetBuilder.cpp	2008-10-18 15:08:39.000000000 +0200
@@ -37,6 +37,7 @@
 #include <geos/geomgraph/Label.h>
 #include <geos/noding/SegmentString.h>
 
+#include <typeinfo>
 #include <cmath>
 #include <vector>
 #include <memory>
diff -urN geos-3.0.0/source/operation/distance/ConnectedElementPointFilter.cpp geos-3.0.0.new/source/operation/distance/ConnectedElementPointFilter.cpp
--- geos-3.0.0/source/operation/distance/ConnectedElementPointFilter.cpp	2007-12-12 22:59:25.000000000 +0100
+++ geos-3.0.0.new/source/operation/distance/ConnectedElementPointFilter.cpp	2008-10-18 15:09:49.000000000 +0200
@@ -19,6 +19,7 @@
 #include <geos/geom/LineString.h>
 #include <geos/geom/Polygon.h>
 
+#include <typeinfo>
 #include <vector>
 
 using namespace std;
diff -urN geos-3.0.0/source/planargraph/DirectedEdge.cpp geos-3.0.0.new/source/planargraph/DirectedEdge.cpp
--- geos-3.0.0/source/planargraph/DirectedEdge.cpp	2007-12-12 22:59:25.000000000 +0100
+++ geos-3.0.0.new/source/planargraph/DirectedEdge.cpp	2008-10-18 15:14:51.000000000 +0200
@@ -18,6 +18,7 @@
 #include <geos/geomgraph/Quadrant.h>
 #include <geos/algorithm/CGAlgorithms.h>
 
+#include <typeinfo>
 #include <cmath>
 #include <sstream>
 #include <vector>
diff -urN geos-3.0.0/source/precision/SimpleGeometryPrecisionReducer.cpp geos-3.0.0.new/source/precision/SimpleGeometryPrecisionReducer.cpp
--- geos-3.0.0/source/precision/SimpleGeometryPrecisionReducer.cpp	2007-12-12 22:59:24.000000000 +0100
+++ geos-3.0.0.new/source/precision/SimpleGeometryPrecisionReducer.cpp	2008-10-18 15:23:34.000000000 +0200
@@ -29,6 +29,7 @@
 #include <geos/geom/LineString.h>
 #include <geos/geom/LinearRing.h>
 
+#include <typeinfo>
 #include <vector>
 
 using namespace std;
diff -urN geos-3.0.0/tests/bigtest/TestSweepLineSpeed.cpp geos-3.0.0.new/tests/bigtest/TestSweepLineSpeed.cpp
--- geos-3.0.0/tests/bigtest/TestSweepLineSpeed.cpp	2007-12-12 22:59:30.000000000 +0100
+++ geos-3.0.0.new/tests/bigtest/TestSweepLineSpeed.cpp	2008-10-18 15:40:00.000000000 +0200
@@ -88,7 +88,7 @@
 //	cout << "n Pts: " << nPts << "   Executed in " << totalTime << endl;
 }
 
-int main(int /* argC */, char* /* argV[] */) {
+int main(int /* argC */, char** /* argV[] */) {
 
 	GeometryFactory *fact=new GeometryFactory();
 
diff -urN geos-3.0.0/tests/xmltester/markup/MarkupSTL.h geos-3.0.0.new/tests/xmltester/markup/MarkupSTL.h
--- geos-3.0.0/tests/xmltester/markup/MarkupSTL.h	2007-12-12 22:59:30.000000000 +0100
+++ geos-3.0.0.new/tests/xmltester/markup/MarkupSTL.h	2008-10-18 15:43:12.000000000 +0200
@@ -62,6 +62,7 @@
 #pragma warning(disable:4786)
 #endif
 
+#include <cstring>
 #include <string>
 #include <map>
 #include <vector>
diff -urN geos-3.0.0/tests/xmltester/XMLTester.cpp geos-3.0.0.new/tests/xmltester/XMLTester.cpp
--- geos-3.0.0/tests/xmltester/XMLTester.cpp	2007-12-21 19:14:09.000000000 +0100
+++ geos-3.0.0.new/tests/xmltester/XMLTester.cpp	2008-10-18 15:41:02.000000000 +0200
@@ -48,6 +48,7 @@
 #include <cassert>
 #include <cctype>
 #include <cstdlib>
+#include <cstring>
 #include <fstream>
 #include <functional>
 #include <iostream>
